allennlp/common/tensor.py

```diff
@@ -44,6 +44,11 @@ def sort_batch_by_length(tensor: torch.FloatTensor, sequence_lengths: torch.Long
     restoration_indices : torch.LongTensor
         Indices into the sorted_tensor such that ``sorted_tensor[restoration_indices] == original_tensor``
     """
+    # Pytorch can't handle sorting with Variables.
+    # TODO(Mark): Assert whether this is allowed - does it break autograd?
```

Comment:
Assert?  You mean figure out?

And yeah, I'm assuming this breaks autograd.  And even if we _can_ do it this way, wouldn't we need to wrap this in a `Variable` when we're done?